sopel/modules/dice.py

```diff
@@ -99,89 +99,153 @@ def get_compressed_string(self):
             dfaces = ("%dx%d" % (times, face) for face, times in dropped)
             dropped_str = "[+%s]" % ("+".join(dfaces),)
 
-        plus_str = ""
-        if self.addition:
-            plus_str = "{:+d}".format(self.addition)
+        return "(%s%s)" % (dice_str, dropped_str)
 
-        return "(%s%s)%s" % (dice_str, dropped_str, plus_str)
-
-    def get_sum(self):
-        """Get the sum of non-dropped dice and the addition."""
-        result = self.addition
+    def get_sum(self) -> int:
+        """Get the sum of non-dropped dice."""
+        result = 0
         for face, times in self.dice.items():
             result += face * times
         return result
 
-    def get_number_of_faces(self):
-        """Returns sum of different faces for dropped and not dropped dice
+    def get_number_of_faces(self) -> int:
+        """Returns sum of different faces for dropped and not dropped dice.
 
-        This can be used to estimate, whether the result can be shown in
-        compressed form in a reasonable amount of space.
+        This can be used to estimate whether the result can be shown (in
+        compressed form) in a reasonable amount of space.
         """
         return len(self.dice) + len(self.dropped)
 
 
-def _roll_dice(bot, dice_expression):
-    result = re.search(
-        r"""
-        (?P<dice_num>-?\d*)
-        d
-        (?P<dice_type>-?\d+)
-        (v(?P<drop_lowest>-?\d+))?
-        $""",
-        dice_expression,
-        re.IGNORECASE | re.VERBOSE)
+class DiceError(Exception):
+    """Custom base exception type."""
+    def __init__(self, *args):
+        super().__init__(*args)
```

Comment:
Is this necessary? It feels like you don't need more than just the class name and the docstring tbh.

Reply:
Hm, I could drop this, yeah. I started out with a defined `message: str` argument (and corresponding property), and `ValueError` as the base class, but clearly things changed and what's left here is useless boilerplate.

Reply:
Ah! I see, I had the same idea: start with a `ValueError` as a base class! :grin:

Reply:
I'm usually okay with `Exception` as the base class when I'm defining a base for a variety of context-specific exceptions, especially if there's any possibility that they might be analogous to different parts of the built-in hierarchy (i.e. a `DiceError` that is _not_ associated with a bad value).

Don't have a strong preference either way here, but I agree that the initializer can go :stuck_out_tongue:

Reply:
I've chosen to keep `DiceError` based on `Exception` instead of `ValueError`, lest the custom subtypes be expected to conform to the semantics of how `ValueError` works (i.e., you construct it with a message). Yeah, I know stuff in plugins is basically internal, but I still want it to be sane-ish.

Regardless, the useless initializer is now gone. 🥳